src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    static final int DATE = 0;
+    static final int NAME = 1;
+    static final int WORKING_HOUR = 2;
+    static final int INCOME_PER_HOUR = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] perHour = new int[names.length];
+        String[] splitedString = new String[4];
+        StringBuilder stringBuilder = new StringBuilder(
+                "Report for period " + dateFrom + " - " + dateTo);
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate parseDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate parseDateTo = LocalDate.parse(dateTo, formatter);
+
+        for (int i = 0; i < names.length; i++) {
+            for (int j = 0; j < data.length; j++) {
+                splitedString = data[j].split(" ");
+                LocalDate parseSplitedDate = LocalDate.parse(splitedString[DATE], formatter);
+                if ((parseSplitedDate.isAfter(parseDateFrom)
+                        || parseSplitedDate.isEqual(parseDateFrom))
+                        && (parseSplitedDate.isBefore(parseDateTo)
+                        || parseSplitedDate.isEqual(parseDateTo))) {
```

Comment:
I think we do not need to check the equality, and `isAfter()` and `isbefore()` are inclusive, please double-check it